### PR TITLE
inference example

### PR DESCRIPTION
**Pull request by rom1504**
computing safety predictions on top of clip embeddings

### CODE REVIEW

**Comment by rom1504** [comment]
https://www.tensorflow.org/guide/data#consuming_python_generators probably would be faster with this
(Already 100k sample/s but could be much faster, it's not using resources well)

File: examples/inference_example.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+import numpy as np
+import fsspec
+import math
+
+def load_safety_model():
+    """load the safety model"""
+    import autokeras as ak  # pylint: disable=import-outside-toplevel
+    from tensorflow.keras.models import load_model  # pylint: disable=import-outside-toplevel
+    from os.path import expanduser  # pylint: disable=import-outside-toplevel
```
Comment (hitchhicker) [inline examples/inference_example.py:13]: This import can be put outside the function, no ?

File: examples/inference_example.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            "https://raw.githubusercontent.com/LAION-AI/CLIP-based-NSFW-Detector/main/clip_autokeras_binary_nsfw.zip"
+        )
+        urlretrieve(url_model, path_to_zip_file)
+        import zipfile  # pylint: disable=import-outside-toplevel
+
+        with zipfile.ZipFile(path_to_zip_file, "r") as zip_ref:
+            zip_ref.extractall(cache_folder)
+
+    loaded_model = load_model(model_dir, custom_objects=ak.CUSTOM_OBJECTS)
+    loaded_model.predict(np.random.rand(10 ** 3, 768).astype("float32"), batch_size=10 ** 3)
```
Comment (hitchhicker) [inline examples/inference_example.py:35]: I guess this line is only used for testing, can be removed.

File: examples/inference_example.py (hunk 1 above)
Comment (rom1504) [reply examples/inference_example.py:13]: Yeah

File: examples/inference_example.py (hunk 2 above)
Comment (rom1504) [reply examples/inference_example.py:35]: It's useful to make sure things are working as expected before the actual inference starts (for example to check it's using the CPU or the GPU), plus it runs XLA JIT a first time, which makes the second call faster

**Comment by rom1504** [comment]
This is WIP.
I'm not even sure I really want to put this here, I need to think some more about it

**Comment by rom1504** [comment]
works better, but still memleak

I think it's due to tf though and not embedding reader

something like https://stackoverflow.com/questions/64199384/tf-keras-model-predict-results-in-memory-leak

**Comment by rom1504** [comment]
works now and no memleak

**Pull request merged**